source/server/GameServer.py

```diff
@@ -82,11 +86,46 @@ def delPlayer(self, player):
 
 
     def nextTurn(self):
-        """Advance to the next trun"""
+        """Advance to the next turn"""
         newIndex = (self.turn_index + 1) % len(self.players)
         self.turn_index = newIndex
         self.players[self.turn_index].Send({"action": "startTurn"})
-        
+        if self.rules.Buy_Option:
+            self.Send_buyingOpportunity()
+
+    def cardBuyingResolution(self):
+        """ Resolve who gets to purchase card for games with Buy_Option = True
+        This is called when upon player drawing cards. """
+        timelimit = time() + self.rules.purchase_time
+        buying_phase = True # else this routine would not be called
+        i_max = len(self.players) - 2
+        index = (self.turn_index + 1) % len(self.players)
+        icount = 0
+        while buying_phase and icount < i_max:
+            while self.players[index].want_card is None and time() < timelimit:
+                # wait patiently(?) while updating information from players.
+                # todo: discuss whether timer should start from time card is discarded, or
+                #  time next player attempts to draw.  Latter would enable players to give one another more time.
+                #  Currently done from time of discard, therefore, may want a longer purchase_time...
```

Comment:
No - currently you call this on draw not on discard - please confirm I am reading "drawWithBuyOption" correctly and update comment

Reply:
Yes-- you are reading it correctly, the comment was wrong (I caught that myself this afternoon).